src/rgw/rgw_auth_s3.cc

```diff
@@ -532,12 +575,12 @@ std::string get_v4_canonical_qs(const req_info& info, const bool using_qs)
 
 boost::optional<std::string>
 get_v4_canonical_headers(const req_info& info,
-                         const std::string& signedheaders,
+                         const boost::string_view& signedheaders,
                          const bool using_qs,
                          const bool force_boto2_compat)
 {
   map<string, string> canonical_hdrs_map;
-  istringstream sh(signedheaders);
+  istringstream sh(signedheaders.to_string());
```

Comment:
TODO: rework the parsing. Kill `istringstream`; use the new variant of `parse_key_value` instead.